src/map/utils/petutils.cpp

```diff
@@ -1531,23 +1550,32 @@ namespace petutils
 
         PPet->loc = PMaster->loc;
 
-        // spawn me randomly around master
-        PPet->loc.p = nearPosition(PMaster->loc.p, CPetController::PetRoamDistance, (float)M_PI);
+        if (petType == PETTYPE_LUOPAN)
+        {
+            PPet->loc.p = nearPosition(PMaster->GetEntity(PMaster->m_TargID)->loc.p, 0.5f, 0.0f);
```

Comment:
I'll admit I haven't _stared_ at luopan spawning on retail; do they actually vary in placement instead of being at the exact location of the target?

Reply:
They actually spawn just behind the target, i guess this is so you can still target the mob, because if the luopan is directly on the mob, its mesess with targeting.